src/Examine.Lucene/Suggest/AnalyzingSuggesterDefinition.cs

```diff
@@ -0,0 +1,76 @@
+using Lucene.Net.Analysis;
+using Lucene.Net.Index;
+using Lucene.Net.Search.Spell;
+using System.Linq;
+using Lucene.Net.Search.Suggest;
+using Lucene.Net.Search.Suggest.Analyzing;
+using Examine.Suggest;
+
+namespace Examine.Lucene.Suggest
+{
+    public class AnalyzingSuggesterDefinition : LuceneSuggesterDefinition, ILookupExecutor
+    {
+        public AnalyzingSuggesterDefinition(string name, string[] sourceFields = null, ISuggesterDirectoryFactory directoryFactory = null, Analyzer queryTimeAnalyzer = null)
+            : base(name, sourceFields, directoryFactory, queryTimeAnalyzer)
+        {
+        }
+
+        /// <inheritdoc/>
+        public Lookup Lookup { get; internal set; }
+
+        /// <inheritdoc/>
+        public override ILookupExecutor BuildSuggester(FieldValueTypeCollection fieldValueTypeCollection, ReaderManager readerManager, bool rebuild)
+            => BuildAnalyzingSuggesterLookup(fieldValueTypeCollection, readerManager, rebuild);
+
+        /// <inheritdoc/>
+        public override ISuggestionResults ExecuteSuggester(string searchText, ISuggestionExecutionContext suggestionExecutionContext) => ExecuteAnalyzingSuggester(searchText, suggestionExecutionContext);
+
+        protected ILookupExecutor BuildAnalyzingSuggesterLookup(FieldValueTypeCollection fieldValueTypeCollection, ReaderManager readerManager, bool rebuild)
+        {
+            string field = SourceFields.First();
```

Comment:
Same here what happens if `SourceFields` are null or empty

Reply:
Added checks to constructor and inline. Changed constructor signature to remove null defaults.